### DISCUSSION TITLE
Pasting long script in bash behaves differently from Terminal.app

### DISCUSSION THREAD

**Discussion by jimmycallin** [ghostty-org/ghostty#3433] [Q&A] [answered]
Hi!

Thank you for ghostty - it's great!

I noticed a different behavior between Terminal.app and Ghostty when pasting multiple long curl scripts (copy pasted from Chrome's network tab). While Terminal.app correctly (albeit slowly) parses my pasted input, Ghostty seem to do something wrong.

Attached is a screen recording of the behavior and the curl script I used. I removed some sensitive information so it will output that the host is not found - that's expected.

zsh seem to work fine. fish breaks down completely unfortunately, but that's at least consistent between Terminal.app and Ghostty.

[curl-paste.txt](https://github.com/user-attachments/files/18260693/curl-paste.txt)

https://github.com/user-attachments/assets/7e93e7ab-5a02-4650-a47d-4e2b808635f2

**Comment by gpanders** [comment, marked as answer]
It looks like bracketed paste mode is not being enabled. When you start bash, open the terminal inspector (Cmd+Option+I) and click the "Modes" tab. Scroll down to the bottom and look for the row labeled `bracketed_paste`. Can you confirm whether or not the box next to it is checked?

If it is not checked, then that would certainly explain the behavior here (although not why it works in Terminal.app, unless Terminal.app _always_ brackets pastes even if the mode is disabled?).

**Reply by taylrfnt** [reply]
To add, in `bash` you can run `bind -V | grep enable-bracketed-paste` to see if bracketed paste is active (I think `Terminal.app` might be acting sneaky in this case)

**Reply by jimmycallin** [reply]
this helped - thanks!

**Comment by taylrfnt** [comment]
I could not reproduce this issue on `tip`, my `bash` is version 5.2, which is newer than some macOS distributions, but bracketed paste was set to on and I saw identical behavior between Ghostty & Terminal.app.  In OP's recording, it looks like the shell is executing each of the dozen or so curl commands as they are pasted/parsed, as far as I can tell
